arch/risc-v/src/mpfs/mpfs_i2c.c

```diff
@@ -622,15 +588,9 @@ static int mpfs_i2c_irq(int cpuint, void *context, void *arg)
 
 static void mpfs_i2c_sendstart(struct mpfs_i2c_priv_s *priv)
 {
-  uint32_t primask;
-
-  primask = mpfs_disable_interrupts();
-
   modifyreg32(MPFS_I2C_CTRL, MPFS_I2C_CTRL_STA_MASK, MPFS_I2C_CTRL_STA_MASK);
 
   up_enable_irq(priv->plic_irq);
```

Comment:
I *think* the irq:s should actually be enabled *before* sending the start. otherwise if there is a pre-emption between modifyreq32 and up_enable_irq, the interrupt is either lost or handled after the pre-emption.

Reply:
that's right